packages/medusa/src/models/publishable-api-key.ts

```diff
@@ -0,0 +1,66 @@
+import { BeforeInsert, Column } from "typeorm"
+
+import { SoftDeletableEntity } from "../interfaces"
+import { resolveDbType } from "../utils/db-aware-column"
+import { generateEntityId } from "../utils"
+import { FeatureFlagEntity } from "../utils/feature-flag-decorators"
+import PublishableAPIKeysFeatureFlag from "../loaders/feature-flags/publishable-api-keys"
+
+@FeatureFlagEntity(PublishableAPIKeysFeatureFlag.key)
+export class PublishableApiKey extends SoftDeletableEntity {
```

Comment:
thought: Do we really want to be able to delete a key and still keep it in the db?

Reply:
This is a good point, probably we don't. I will replace it with BaseEntity.